public/app/core/logsModel.ts

```diff
@@ -474,10 +474,11 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
       kind: LogsMetaKind.LabelsMap,
     });
   }
-
-  const limits = logSeries.filter((series) => series.meta && series.meta.limit);
+  // Data sources that set up searchWords on backend use meta.custom.limit
+  // Data sources that set up searchWords trough frontend can use meta.limit
```

Comment:
Nitpick (=you can ignore):

```suggestion
  // Data sources that set up searchWords on backend use meta.custom.limit.
  // Data sources that set up searchWords through frontend can use meta.limit.
```